bigquery_etl/query_scheduling/task.py

```diff
@@ -418,6 +428,21 @@ def _duplicate_dependency(task):
                             dependencies.append(task)
                         break  # stop after the first match
```

Comment:
The `*_stable.*` pattern is intended to match last. All the other stable table dependencies do depend on `copy_deduplicate_all`. So if one of `copy_deduplicate_all` downstream dependencies matches then it is unnecessary for `copy_deduplicate_all` to be listed as dependency as well.
I think over time we have been adding more external dependencies so I would probably be worth reconsidering, or treating `copy_deduplicate_all` as a special case. I'll open a separate ticket since that is outside of the scope of this PR

Reply:
> The `*_stable.*` pattern is intended to match last.

The `*.clients_last_seen_joined` pattern being added in this PR is currently the last pattern, so should it be moved before the `*_stable.*` pattern then?

Reply:
Indeed. Fixed and comment added.